docs/proposals/auth-gateway.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Authentication Gateway"
+linkTitle: "Authentication Gateway"
+weight: 1
+slug: "auth-gateway"
+---
+
+- Author: [Doğukan Teber](https://github.com/dogukanteber)
+- Date: March 2023
+- Status: Proposed
+
+## Overview
+
+If you run Cortex for multiple tenants you need to identify your tenants every time they send metrics or query them. This is needed to ensure that metrics can be ingested and queried separately from each other. For this purpose, the Cortex microservices require you to pass a Header called `X-Scope-OrgID`. Unfortunately, the Prometheus Remote write API has no config option to send headers and for Grafana you must provide a data source to do so. Therefore the Cortex k8s manifests suggest deploying an NGINX server inside of each tenant which acts as a reverse proxy. Its sole purpose is proxying the traffic and setting the `X-Scope-OrgID` header for your tenant.
+
+## Proposal
+
+We propose to solve this problem by adding a gateway that can be considered the entry point for all requests towards Cortex. This gateway will be responsible for handling multi-tenancy features without complicated proxy configuration.
+
+This gateway will be located in another repository, [cortexproject/auth-gateway](https://github.com/cortexproject/auth-gateway). For the first version, basic authentication will be supported. However, as the project grows, it is planned to support different authentication mechanisms according to the needs of the users.
+
+We would like to have a simple configuration format. The following is the proposed configuration:
+
+```yaml
+server:
+  address: localhost
+  port: 8080
+admin:
+  address: localhost
+  port: 8090
+tenants:
+  - authentication: basic
+    username: username1
+    password: password1
+    id: orgid
+
+  - authentication: basic
+    username: username2
+    password: password2
+    id: orgid
+distributor:
+  dns_refresh_interval: 3s
+  url: http://distributor:9009
+  paths:
+  - /api/v1/push
+  - /api/prom/push
+  http_client_timeout: 20
```

Comment:
Dogukan wanted to do it how Alvin suggested it. I discourage it, because I know with too many levels it becomes hard to read.
I was looking at https://cortexmetrics.io/docs/configuration/configuration-file/#server_config when I suggested the options you see in the yaml now (http_client_timeout, etc.). 

I still support what I suggested. I am ok with whatever we decide as a group.  what is your vote? @dogukanteber @alvinlin123

Reply:
I am honestly okay with both of the approaches. I understand @alvinlin123's perspective and his way of doing would be more structured. However, @friedrichg is also right to avoid nested configuration.

I vote for the current proposed implementation but as Friedrich said, I am okay with whatever we decide as a group.

Reply:
I see @friedrichg's reasoning make sense too. I don't have too strong of an opinion here. My strongest reason for my suggestion was for consistency with existing Cortex configuration style, but  auth-gateway is going to be in separate repo, so this reason is not strong anymore. 

Let's go with the option with least amount of work, which is do nothing and leave the current design as-is.

Reply:
Okay. I am assuming we agreed on the proposed design.

Reply:
Is the `http verb` missing on those paths? Usually when we configure paths we put the verb as well (`GET`, `Post` etc)..

Also, im assuming this for now will be a simple string matching, right? We will not be able to do something like `/prefix/{var1}/sufix` as of now?

Reply:
@alanprot you are right, we should get the verb there as well somehow

it's more like a prefix match only, no complex one